File: BestTimeToBuyAndSellStock(121).js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = prices[0];
    let profit = 0;
    
    for(let i=1; i<prices.length; i++) {
        if(prices[i]<min) {
            min = prices[i];
        }
        if(prices[i]-min > profit) {
            profit = prices[i]-min;
        }
    }
    
    return profit;
};

File: BestTimeToBuyAndSellStock(121).swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        if prices.count == 1 {
            return 0
        }

        var res = 0
        var left = 0
        for right in 1..<prices.count {
            if prices[right]-prices[left] < 0 {
                left = right
            } else if prices[right]-prices[left] > res {
                res = prices[right]-prices[left]
            }
        }

        return res
    }
}
